### DISCUSSION TITLE
Multiple input fields in note filter

### DISCUSSION THREAD

**Discussion by mortii** [mortii/anki-morphs#67] [Ideas]
As talked about in #45

**Comment by mortii** [comment]
Sibling cards are basically the strongest argument for having multiple input fields right? I have essentially zero sibling cards so this never crossed my mind.

**Reply by Vilhelm-Ian** [reply]
Yes it's a pretty valid reason. I also like you have zero sibligs. So I don't have motivation to think about the problem. To me it seems like it will add a lot of complexity

**Reply by mortii** [reply]
With regards to using reversed siblings:
If you are at the point where the question and the answer is in the same language, then AnkiMorphs has probably passed the point of diminishing returns for you in that particular language.  I think a large part of designing applications is striking the right balance between implementing potentially useful features that creates further complexity & protecting the users from themselves by using said features to their detriment.

**Reply by Vilhelm-Ian** [reply]
I just reread the previous thread. I misunderstood how aleks was using multiple fields.  I am quit suprised that **two** people thought of using that option. I think we should go in the opposite direction and prevent a person from creating two filters with the same name to discourage people from misusing the addon.

**Comment by mortii** [comment]
>> One of the main reasons I changed it to only be one field is that we could now have a drop down menu of the fields instead of typing them manually. I would always misspell the field name or miss an upper-case letter and the entire filter would then be useless.

>IIRC, such typos cause an error in MorphMan, although maybe the error happens too late.

_Originally posted by @aleksejrs in https://github.com/mortii/anki-morphs/issues/45#issuecomment-1819855999_

Not once did MorphMan catch that I typed "japanese" with a lowecase letter instead of "Japanese" with an upper case like the field actually had.

If we want to support multiple input fields then maybe they could be auto-completed and checked like tags are in the anki browser. I also propose doing something like that for the tags selection in the note filter( #61).

**Reply by aleksejrs** [reply]
I am not sure I've ever made a case mistake, but I remember that my old MorphMan crashes when trying to recalc (probably at the second stage) with the wrong fields. But it could just be after deleting a note type, unless it just removed it in that case?

**Reply by aleksejrs** [reply]
I am really disappointed at how the usability of "Grid Areas" in [Grid Fields](https://ankiweb.net/shared/info/796923308), where you have to specify *all* the fields, compares to [Multi-column note editor](https://ankiweb.net/shared/info/3491767031), where you could just click a field to make it take the whole row. I am not blaming the developer for not being able to implement it properly, it's just that it is unusable for me.

Fortunately, I am good enough at copying text for MorphMan. Also, most of my fields to be analyzed are named "1", "2", "Text1", "Text2", "myalttext", "Word", "Name".

**Reply by Vilhelm-Ian** [reply]
I really like the dropdown method it makes it much more user friendly. A big critque of MorphMan was how complicated it is. https://youtu.be/DfSfPS-Nx7Q?t=1311 just watch 20 seconds of this clip

**Reply by Vilhelm-Ian** [reply]
> I am really disappointed at how the usability of "Grid Areas" in [Grid Fields](https://ankiweb.net/shared/info/796923308), where you have to specify all the fields, compares to [Multi-column note editor](https://ankiweb.net/shared/info/3491767031),

@aleksejrs can you provide a screenshot and add a arrow I don't get which part you are talking about

**Reply by aleksejrs** [reply]
@Vilhelm-Ian, Multi-column note editor arranges all the fields in n columns, but the "«-»" button expands the field to span across all columns.
Grid Fields, of which even [the newer version](https://ankiweb.net/shared/info/1796231164) didn't work for me (I think there was a newer patch somewhere, unless it was that one), supports either a number of columns, or you can make a table of the fields as text. But it says you have to list **all** fields, which is very hard with all the text and flag and MorphMan fields that I have.
So that's kind of an extreme version of the problem mortii has.

**Reply by Vilhelm-Ian** [reply]
![image](https://github.com/mortii/anki-morphs/assets/81630959/53a4b1c6-e491-4d9b-b569-04ee268a0147)
The problem is that this has too many entries? So you want it to display like the addons above

**Reply by aleksejrs** [reply]
@Vilhelm-Ian, I am talking about the problem that that dropdown solves for mortii. I guess I shouldn't have mentioned it; it confused you.

**Reply by mortii** [reply]
> @Vilhelm-Ian, Multi-column note editor arranges all the fields in n columns, but the "«-»" button expands the field to span across all columns.
Grid Fields, of which even [the newer version](https://ankiweb.net/shared/info/1796231164) didn't work for me (I think there was a newer patch somewhere, unless it was that one), supports either a number of columns, or you can make a table of the fields as text. But it says you have to list all fields, which is very hard with all the text and flag and MorphMan fields that I have.
So that's kind of an extreme version of the problem mortii has.

Those are interesting add-ons. I want the settings dialog interface to be as small as possible (and as few columns as possible), so I don't know if those approaches are are the best ways to go.

**Reply by Vilhelm-Ian** [reply]
@mortii for which part of the settings dialogue do you want to make it more compact. The Note Filters and the Extra fields? I don't see them being able to be displayed in multiple collums since they are tabular. The other tabs already fit on one screen so I don't see them benefiting other than being more aesthethic

**Reply by mortii** [reply]
I don't like the note filters being so wide (because of all the columns), right now I'd say it is on the edge of acceptable, but if we were to introduce more columns then a redesign might be warranted.

**Comment by mortii** [comment]
Having multiple input fields definitely has valid use cases. That being said I think those use cases are too narrow for the complexity it will bring to the average user, but most of all the complexity it will add to the code base.

I think one of the biggest problems MorphMan has is that it is too complicated, and I don't want to the same thing to happen with AnkiMorphs.

I really value honest and constructive feedback and discussions--having your ideas and solutions challenged is crucial to finding the best antifragile ideas and solutions.

Thanks for providing that feedback, and apologies for not being willing to implement the feature :pray:
